src/pl/senla/task9/ex3/Factory.java

```diff
@@ -0,0 +1,50 @@
+package pl.senla.task9.ex3;
+
+import java.util.ArrayDeque;
+import java.util.Queue;
+
+import static java.lang.Thread.*;
+import static pl.senla.task9.ex3.Constant.MAX_BUFFER_CAPACITY;
+
+public class Factory {
+
+    private final Queue<Integer> buffer = new ArrayDeque<>();
```

Comment:
Think about ArrayBlockingQueue, coz it's part of concurrent package which is mor suitable for thread related tasks.

Reply:
Or any other queue from concurrent package

Reply:
ready.
now it looks much better.
Thank you for advice.